<?xml version='1.0'?>
<robot xmlns:xacro="http://wiki.ros.org/xacro">
    <xacro:macro name="ros2_control_system">
        <ros2_control name="imu" type="sensor">
            <hardware>
                <plugin>psd_vehicle_hardware_interfaces/psd_vehicleImuSensor</plugin>
                <param name="connection_timeout_ms">120000</param>
                <param name="connection_check_period_ms">500</param>
            </hardware>
            <sensor name="imu">
                <state_interface name="orientation.x" />
                <state_interface name="orientation.y" />
                <state_interface name="orientation.z" />
                <state_interface name="orientation.w" />
                <state_interface name="angular_velocity.x" />
                <state_interface name="angular_velocity.y" />
                <state_interface name="angular_velocity.z" />
                <state_interface name="linear_acceleration.x" />
                <state_interface name="linear_acceleration.y" />
                <state_interface name="linear_acceleration.z" />
            </sensor>
        </ros2_control>

        <ros2_control name="wheels" type="system">
            <hardware>
                <plugin>psd_vehicle_hardware_interfaces/psd_vehicleSystem</plugin>
                <param name="connection_timeout_ms">120000</param>
                <param name="connection_check_period_ms">500</param>

                <!-- order of velocity commands to be published in motors_cmd Float32MultiArray msg -->
                <param name="velocity_command_joint_order">
                    drive_rr,
                    drive_rl,
                    drive_fr,
                    drive_fl
                </param>
            </hardware>

            <joint name="drive_fl">
                <command_interface name="velocity" />
                <state_interface name="position" />
                <state_interface name="velocity" />
            </joint>
            <joint name="drive_fr">
                <command_interface name="velocity" />
                <state_interface name="position" />
                <state_interface name="velocity" />
            </joint>
            <joint name="drive_rl">
                <command_interface name="velocity" />
                <state_interface name="position" />
                <state_interface name="velocity" />
            </joint>
            <joint name="drive_rr">
                <command_interface name="velocity" />
                <state_interface name="position" />
                <state_interface name="velocity" />
            </joint>
        </ros2_control>
    </xacro:macro>
</robot>